File: sorting_algorithms/insertion_sort/insertion_sort.go
package main

import "fmt"

func insertion_sort(A []int) []int {
	for i := 1; i < len(A); i++ {
		hole := i
		value := A[i]

		for {
			if hole <= 0 || A[hole-1] < value {
				break
			}
			A[hole] = A[hole-1] // shifting greater vlaue to right
			hole = hole - 1
		}
		A[hole] = value // shifting value to sorted place
	}

	return A
}

func main() {
	t := insertion_sort([]int{7, 2, 5, 3, 1})
	fmt.Println(t)
}

File: sorting_algorithms/insertion_sort/insertion_sort.py
def insertion_sort(A):
    for i in range(1,len(A)):
        hole = i
        value = A[i]

        while (hole > 0 and A[hole -1] > value):
            # shifting greater values to the right
            A[hole] = A[hole - 1]
            hole = hole - 1
        # putting the value in its sorted place
        A[hole] = value

t = [7,2,5,3,1]
insertion_sort(t)
print(t)